core/metacling/src/TClingCallFunc.cxx

```diff
@@ -1389,384 +1389,85 @@ TClingCallFunc::make_dtor_wrapper(const TClingClassInfo *info)
-               break;
-            case BuiltinType::UChar: vh.u.uc = fArgVals[i].simplisticCastAs<unsigned char>();
-               break;
-            case BuiltinType::WChar_U: vh.u.wc = fArgVals[i].simplisticCastAs<wchar_t>();
-               break;
-            case BuiltinType::Char16: //vh.u.c16 = fArgVals[i].simplisticCastAs<char16_t>();
-               break;
-            case BuiltinType::Char32: //vh.u.c32 = fArgVals[i].simplisticCastAs<char32_t>();
-               break;
-            case BuiltinType::UShort: vh.u.us = fArgVals[i].simplisticCastAs<unsigned short>();
-               break;
-            case BuiltinType::UInt: vh.u.ui = fArgVals[i].simplisticCastAs<unsigned int>();
-               break;
-            case BuiltinType::ULong: vh.u.ul = fArgVals[i].simplisticCastAs<unsigned long>();
-               break;
-            case BuiltinType::ULongLong: vh.u.ull = fArgVals[i].simplisticCastAs<unsigned long long>();
-               break;
-               //  Signed Types
-            case BuiltinType::Char_S: vh.u.c = fArgVals[i].simplisticCastAs<char>();
-               break;
-            case BuiltinType::SChar: vh.u.sc = fArgVals[i].simplisticCastAs<signed char>();
-               break;
-            case BuiltinType::WChar_S: vh.u.wc = fArgVals[i].simplisticCastAs<wchar_t>();
-               break;
-            case BuiltinType::Short: vh.u.s = fArgVals[i].simplisticCastAs<short>();
-               break;
-            case BuiltinType::Int: vh.u.i = fArgVals[i].simplisticCastAs<int>();
-               break;
-            case BuiltinType::Long: vh.u.l = fArgVals[i].simplisticCastAs<long>();
-               break;
-            case BuiltinType::LongLong: vh.u.ll = fArgVals[i].simplisticCastAs<long long>();
-               break;
-            case BuiltinType::Float: vh.u.flt = fArgVals[i].simplisticCastAs<float>();
-               break;
-            case BuiltinType::Double: vh.u.dbl = fArgVals[i].simplisticCastAs<double>();
-               break;
-            case BuiltinType::LongDouble: vh.u.ldbl = fArgVals[i].simplisticCastAs<long double>();
-               break;
-            case BuiltinType::NullPtr: vh.u.vp = fArgVals[i].getPtr();
-               break;
-            default:
-               // Thee should be no others.  This is here in case
-               // this changes in the future.
-               ::Error("TClingCallFunc::exec(void*)",
-                       "Unhandled builtin type '%s'",
-                       BT->getTypeClassName());
-                  QT->dump();
-               return;
-            }
-            vh_ary.push_back(vh);
-            vp_ary.push_back(&vh_ary.back());
-         } else if (QT->isReferenceType() || QT->isRecordType()) {
+      for (unsigned i = 0; i < num_args; ++i) {
+         QualType QT;
+         // Check if we provided a this parameter.
+         // FIXME: Currently we do not provide consistently the this pointer as
```

Comment:
Can you be more specific on what "provide" means? We need to provide the `this` pointer I suppose, only it can be left untouched and default initialized to `nullptr` - is that what you mean?